### ISSUE TITLE
Error in thread.h found be the cppcheck engine

### ISSUE THREAD

**Issue by mrGoodWolf** [minetest/minetest#13838]
### Minetest version

```true
source codefrom ten minutes ago (as of posting)
```

### Active renderer

threading error

### Irrlicht device

none

### Operating system and version

macos Monterey 12.6.8.9.2

### CPU model

Apple Intel

### Summary

i Was testing source with cppcheck when the first item was a fetal error saying some macro was unknown.

### Steps to reproduce

this is the command i used
```bash
$brew install cppcheck

==> Downloading https://formulae.brew.sh/api/formula.jws.json
######################################################################### 100.0%
==> Downloading https://formulae.brew.sh/api/cask.jws.json
######################################################################### 100.0%
==> Downloading https://ghcr.io/v2/homebrew/core/cppcheck/manifests/2.12.1
######################################################################### 100.0%
==> Fetching dependencies for cppcheck: pcre and tinyxml2
==> Downloading https://ghcr.io/v2/homebrew/core/pcre/manifests/8.45
######################################################################### 100.0%
==> Fetching pcre
==> Downloading https://ghcr.io/v2/homebrew/core/pcre/blobs/sha256:5e5cc7a5bf8bb
######################################################################### 100.0%
==> Downloading https://ghcr.io/v2/homebrew/core/tinyxml2/manifests/9.0.0
######################################################################### 100.0%
==> Fetching tinyxml2
==> Downloading https://ghcr.io/v2/homebrew/core/tinyxml2/blobs/sha256:24e501c80
######################################################################### 100.0%
==> Fetching cppcheck
==> Downloading https://ghcr.io/v2/homebrew/core/cppcheck/blobs/sha256:c28d5b930
######################################################################### 100.0%
==> Installing dependencies for cppcheck: pcre and tinyxml2
==> Installing cppcheck dependency: pcre
==> Downloading https://ghcr.io/v2/homebrew/core/pcre/manifests/8.45
Already downloaded: /Users/trey/Library/Caches/Homebrew/downloads/4a3bba0e8d1899c13ac3442d2c49df6c1999948bdc0943f2f179693257d82545--pcre-8.45.bottle_manifest.json
==> Pouring pcre--8.45.monterey.bottle.tar.gz
🍺  /usr/local/Cellar/pcre/8.45: 204 files, 5.7MB
==> Installing cppcheck dependency: tinyxml2
==> Downloading https://ghcr.io/v2/homebrew/core/tinyxml2/manifests/9.0.0
Already downloaded: /Users/trey/Library/Caches/Homebrew/downloads/b01d9308a0dd7632eaccb55c3bbe682aa372409c5ac197729d01dca8e499a2f9--tinyxml2-9.0.0.bottle_manifest.json
==> Pouring tinyxml2--9.0.0.monterey.bottle.tar.gz
🍺  /usr/local/Cellar/tinyxml2/9.0.0: 13 files, 204.7KB
==> Installing cppcheck
==> Pouring cppcheck--2.12.1.monterey.bottle.tar.gz
🍺  /usr/local/Cellar/cppcheck/2.12.1: 83 files, 10.3MB
==> Running `brew cleanup cppcheck`...
Disable this behaviour by setting HOMEBREW_NO_INSTALL_CLEANUP.
Hide these hints with HOMEBREW_NO_ENV_HINTS (see `man brew`).
$cppcheck --enable=warning --inconclusive --library=posix --dump *.cpp 
threading/thread.h:61:2: error: There is an unknown macro here somewhere. Configuration is required. If DISABLE_CLASS_COPY is a macro then please configure it. [unknownMacro]
 DISABLE_CLASS_COPY(Thread)
 ^
```
it happened more that once but i think all in the same file

**Comment by appgurueu** [comment]
Please read the "error" message carefully:

```
$ cppcheck --enable=warning --inconclusive --library=posix --dump *.cpp 
threading/thread.h:61:2: error: There is an unknown macro here somewhere. Configuration is required. If DISABLE_CLASS_COPY is a macro then please configure it. [unknownMacro]
 DISABLE_CLASS_COPY(Thread)
 ^
 ```
 
This is *not* an "error in `threading/thread.h`". `cppcheck` is missing configuration here; the `DISABLE_CLASS_COPY` macro is defined in `util/basic_macros.h`, which is included by `threading/thread.h`. I have no idea why `cppcheck` fails to understand this.
 
**TL;DR:** This is an (IMO low priority until a case has been made for `cppcheck`) feature request ("add cppcheck configuration") at best. Definitely not a bug. This issue should be closed as invalid or retagged & renamed to a (low-priority) feature request.
